qip-0006.md

```diff
@@ -0,0 +1,512 @@
+processing speed of an individual computer thread, due to the sequential nature
+of the SDF. The total number of threads or computers is irrelevant; what
+matters is the time it takes for a single thread to run the SDF.
+
+To establish this relationship, the network needs to continuously measure and
+adjust the conversion from delay time to the number of SDF rounds. This calls
+for a mechanism that incentivizes miners to accurately gauge and report their
+hashing efficiency. One way to implement this is by requiring that the nonce of
+a block be valid only if it falls within a specific range. This range is
+determined by the output of an SDF run, which is included in the block header.
+
+The same SDF used for creating MMTX bitshares would apply here. However, a
+unique feature of this SDF is the creation of a periodic hash manifest at the
+end of each run. This manifest, recording 1024 evenly spaced intermediate
+hashes, allows for faster validation of the block header – up to 1024 times
+quicker than the actual clock-time. This makes the SDF function effectively a
+Verifiable Delay Function (VDF) in this context, a feature that will be
+referred to as the "SDF Challenge".
+
+The SDF Challenge will generate a series of measurements that help to
+continuously calibrate the conversion of delay time to the number of rounds,
+according to the formula:
+
+$$ n = \kappa*d_{SDF,i,j} $$
+
+Here, $n$ is the number of SDF rounds required to achieve the expected delay
+$d_{SDF,i,j}$, and $\kappa$ is a conversion factor indicating iterations per
+second.
+
+To accommodate this SDF Challenge, modifications to the block header are
+necessary. This includes adding a commitment to the SDF Challenge and removing
+three fields: parentHash[2], Number[2], and Nonce.
+
+#### Modified Header Fields ####
+
+| Field            | Type/Value       |
+|------------------|------------------|
+| ParentHash[0]    | common.Hash      |
+| ParentHash[1]    | common.Hash      |
+| UncleHash        | common.Hash      |
+| Coinbase         | common.Address   |
+| Root             | common.Hash      |
+| TxHash           | common.Hash      |
+| EtxHash          | common.Hash      |
+| EtxRollupHash    | common.Hash      |
+| ManifestHash[0]  | common.Hash      |
+| ManifestHash[1]  | common.Hash      |
+| ManifestHash[2]  | common.Hash      |
+| ReceiptHash      | common.Hash      |
+| Number[0]        | *big.Int         |
+| Number[1]        | *big.Int         |
+| GasLimit         | uint64           |
+| GasUsed          | uint64           |
+| BaseFee          | *big.Int         |
+| Difficulty       | *big.Int         |
+| Location         | []byte           |
+| Time             | uint64           |
+| Extra            | []byte           |
+| SDFChallenge     | common.Hash      |
+
+### SDF Challenge ###
+
+The initial SDF challenge hash will be created by taking the
+$Hash(parentHash[2],root)$. A miner can choose to sample the SDF challenge at
```

Comment:
you say `parentHash[2]`, but the header only shows `parentHash[0]` & `parentHash[1]` as fields. Same for number. Is this a mistake?

Reply:
Not a mistake.  The things that are taken out of the header are added to the bitshare. This is done because a transaction needs to be able to reference these specific fields and not any of the other fields in the header.

Reply:
Your equation uses parentHash[2], but it isn't defined anywhere.

Reply:
outdated